opentelemetry-sdk/src/opentelemetry/sdk/metrics/__init__.py

```diff
@@ -0,0 +1,241 @@
+# Copyright 2019, OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+from typing import Sequence, Tuple, Type
+
+from opentelemetry import metrics as metrics_api
+
+logger = logging.getLogger(__name__)
+
+
+class BaseHandle:
+    def __init__(
+        self,
+        value_type: Type[metrics_api.ValueT],
+        enabled: bool,
+        monotonic: bool,
+    ):
+        self.data = 0
+        self.value_type = value_type
+        self.enabled = enabled
+        self.monotonic = monotonic
+
+    def _validate_update(self, value: metrics_api.ValueT):
+        if not self.enabled:
+            return False
+        if self.monotonic and value < 0:
```

Comment:
Shouldn't a monotonic gauge prevent you from setting a `value` less than the current value instead?

Reply:
You are correct. Fixed.